src/pages/index.js

```diff
@@ -80,6 +56,165 @@ const cardLinkInput = addCardFormElement.querySelector(
+      });
+  }
+}
+
+// Edit Avatar
+
+const editAvatarPopup = new PopupWithForm(
+  "#avatar-edit-modal",
+  handleEditAvatar,
+  "Save"
+);
+
+function handleEditAvatar(data) {
+  editAvatarPopup.setLoading(true);
+  api
+    .updateProfileAvatar(data.link)
+    .then((userData) => {
+      userInfo.setUserAvatar(userData.avatar);
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopup.setLoading(false, "Saving");
+    });
+}
+
+editAvatarPopup.setEventListeners();
+
+profileAvatarButton.addEventListener("click", () => editAvatarPopup.open());
+
+// User Info + Update Profile
+
+const userInfo = new UserInfo(profileTitle, profileDescription, avatarElement);
+
+function fillProfileForm() {
+  const userInfoData = userInfo.getUserInfo();
+  profileTitleInput.value = userInfoData.name;
+  profileDescriptionInput.value = userInfoData.description;
+}
+
+api.getUserInfo().then((userData) => {
+  userInfo.setUserInfo({
+    name: userData.name,
+    description: userData.about,
+  });
+});
+
+function handleProfileEditSubmit(data) {
+  editProfilePopup.setLoading(true);
+  api
+    .updateProfile(data)
+    .then((userData) => {
+      userInfo.setUserInfo({
+        name: userData.name,
+        description: userData.about,
+      });
+      fillProfileForm();
+      editProfilePopup.close();
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => {
+      editProfilePopup.setLoading(false);
+    });
+}
+
+const editProfilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit,
+  "Save"
+);
+editProfilePopup.setEventListeners();
+
+// Add Card APi
+
+function handleAddCardFormSubmit() {
+  const { value: name } = cardTitleInput;
+  const { value: link } = cardLinkInput;
+  const cardData = { name, link };
+
+  addCardPopup.setLoading(true);
+
+  api
+    .addCard(cardData)
+    .then((res) => {
+      renderCard(res, cardListEL);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardPopup.setLoading(false, "Create");
+    });
+}
+
+const addCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit,
+  "Create"
+);
+addCardPopup.setEventListeners();
+
+function openProfileForm() {
+  fillProfileForm();
+  cardListEL;
+  editProfilePopup.open();
+}
+
+//Delete Card const
+
+const deleteCardPopup = new PopupWithConfirmation(
+  "#card__delete-modal",
+  "Deleting..."
+);
+
+deleteCardPopup.setEventListeners();
+
+function handleDeleteCardClick(cardId, card) {
+  deleteCardPopup.setSubmitAction(() => {
+    api
+      .removeCardOnServer(cardId)
+      .then(() => {
+        card.removeCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {});
+  });
+  deleteCardPopup.open();
+}
+
 /*Form Validation*/
 
 const settings = {
```

Comment:
From the 8th sprint all constants (`initialCards` and `settings`) should be in file `constants.js` in folder `utils` to clean up the main file.